tools/kubernetes/helm/hue/values.yaml

```diff
@@ -45,6 +45,19 @@ websocket:
   replicas: 1
 monitoring:
   enabled: false
+alerting:
+  enabled: false
+  prometheusRules:
+    - name: hue-pod-monitor.alert.rules
+      rules:
+      - alert: hue_pods_are_down
+        expr: count(up{job="hue"})==0 #Update the rule according to alerting requirements
+        for: 30m #Change this based on your SLA
+        labels:
+          severity: High
+        annotations:
+          summary: Hue pods are down
+          impact: Users might not be able to query via Hue Web UI
```

Comment:
This could live in `prometheus-rule-hue.yaml` to be simpler?
(maybe just `30m` as a variable, and we could remove the comments)

Reply:
**Why move the rule to values.yaml ?** 

You are right , Initially I thought the rules are better off in _prometheus-rule-hue.yaml_ to have a simpler values.yaml. 
But released that any one enabling alerting shall have to tweak the template to make changes in the rules. 
I think it's not good to touch the templates directly especially for changing the rules because this could easily source code in the same organization. 

The above was experienced by me, when using a staging/prod setup. Usually the team would end up with a different value files (value-staging.yaml and value-prod.yaml) in such cases rules could be tweaked easily from the values file itself rather than making changes in templates (If they edit the template needless to say staging/prod have a code drifted already)

Hope the above reasoning is valid enough. 


**Changing for: 30m to variable like sla:30**

If the above change of having the rules in value.yaml is good then using variable here would not be that helpful I think. 

But apart from this , I had a look into variable declaration inside the Prometheus rule and found out that the rule gets directly translated into a file and gets mounted (hue-pod-monitor.rules), hence inside this code block _prometheusRules_ I couldn't find a easier way to variable replacement. Anyway such attempt resulted in a prometheus exceptions says alert rule is not valid. 

Considering the attempt here is to enable alerting rule and with a sample example I was wondering if its okay to leave this like this ?

Reply:
Interesting, what were the differences of stagings vs prod? 
(trying to see if we can't still refactor to a lower denominator as that feels weird so much yaml as values)

Reply:
Example : Staging hue is connected to staging Impala , Hive and few DB's and runs in staging namespace 
Prod the same (respective prod Impala etc) . In this case we have 2 values fine we maintain and the template remains the same.

Reply:
Yes, but related to alerting?

Not sure why this would be different except "maybe" the sla?
expr: count(up{job="hue"})==0

Reply:
I think , I get your point. 
I am was a bit focused on how my current setup where-in a single K8's cluster is having both staging/prod namespace. Technically this isn't a case everywhere and this can definitely vary. 
I can move this back to prometheus-rule-hue.yaml.

Reply:
I need some input on **Changing for: 30m to variable like sla:30** thing though, I couldn't really find a good way to make variable replacement inside the rule block. Do you think we can still keep **for: 30m** as a example ? (I can definitely spend a bit more time to see if this could be achieved )

Reply:
:+1: Always good to keep the content of values as simple as possible :)

If this does not work it is also good to keep it hardcoded for now! (we can always iterate)
`for: {{ .Values.alerting.sla }}`

Reply:
Noted 👍 . I will test this (and one more variable declaration comment above) and shall update the PR respectively. Thanks for your quick comments/reviews

Reply:
So after a few testing I can confirm that variable declaration inside the Prometheus rule is not that straight forward as I thought. All types of variable declaration (Inside template , from values file etc) fail with any error like/similar to below 

`Error: UPGRADE FAILED: template: hue/templates/prometheus-rule.yaml:16:21: executing "hue/templates/prometheus-rule.yaml" at <.Values.alerting.sla>: nil pointer evaluating interface {}.sla` 

Which leads me to say/think that the complete section of rules is directly taken from template and stored in a file for Prometheus to pick. 

If you agree with iterating process then we could leave it hardcoded for now.

Reply:
:+1: Thanks for testing and let's keep it hardcoded in the rule.yaml (and remove the `prometheusRules` piece in values.yaml)!